src/main/java/blang/distributions/Generators.java

```diff
@@ -80,7 +81,19 @@ public static double beta(Random random, double alpha, double beta)
       result = ONE_MINUS_EPS;
     return result;
   }
-  
+    public static int BetaBinomial(Random random, double alpha, double beta, int numberOfTrials)
+    {
+        double x = beta(random, alpha, beta);
+        int y = binomial(random,numberOfTrials,x);
+        return y;
+    }
+    public static int hyperGeometric(Random random, int numberOfDraws, int population, int populationGood) {
+        int result = new HypergeometricDistribution(generator(random), population, populationGood, numberOfDraws).sample();
+        if(result==0.0) {
```

Comment:
We don't need the check result == 0.0 here.

These checks are when a continuous distribution is used and there is an asymptote at one of the boundaries. Here the distribution is discrete. 

It's a bit weird to compare an int and a double like that too.